pydantic_ai_slim/pydantic_ai/providers/cerebras.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations as _annotations
+
+import os
+from typing import overload
+
+from httpx import AsyncClient as AsyncHTTPClient
+from openai import AsyncOpenAI
+
+from pydantic_ai.exceptions import UserError
+from pydantic_ai.models import cached_async_http_client
+from pydantic_ai.profiles import ModelProfile
+from pydantic_ai.profiles.google import GoogleJsonSchemaTransformer
+from pydantic_ai.profiles.openai import OpenAIModelProfile
+from pydantic_ai.providers import Provider
+
+try:
+    from openai import AsyncOpenAI
+except ImportError as _import_error:  # pragma: no cover
+    raise ImportError(
+        'Please install the `openai` package to use the Cerebras provider, '
+        'you can use the `openai` optional group — `pip install "pydantic-ai-slim[openai]"`'
+    ) from _import_error
+
+
+class CerebrasProvider(Provider[AsyncOpenAI]):
+    """Provider for Cerebras API."""
+
+    @property
+    def name(self) -> str:
+        return 'cerebras'
+
+    @property
+    def base_url(self) -> str:
+        return 'https://api.cerebras.ai/v1'
+
+    @property
+    def client(self) -> AsyncOpenAI:
+        return self._client
+
+    def model_profile(self, model_name: str) -> ModelProfile | None:
```

Comment:
I think we should do automatic model profile selection here based on the underlying model used, like we do in `TogetherProvider`. We can still override the `json_schema_transformer` unconditionally, like we do there as well.

Reply:
I'm a bit puzzled on 
```python
        provider, model_name = model_name.split('/', 1)
        if provider in provider_to_profile:
            profile = provider_to_profile[provider](model_name)
```

I dont fully grasp the idea behind what's meant to be a "provider" in this case. Should CerebrasProvider support alternative model name syntax like `Qwen/Qwen-3-32b` instead of what's listed in their docs: `qwen-3-32b`?

Reply:
@smallstepman What we're trying to do is to use the regular model names that Cerebras expects to choose the right model profile for the model in question, based on the prefix. In the case of `TogetherProvider` their model names have a `<provider>/` prefix, but for example in `GroqProvider` you can see how we match on a simple `qwen` prefix and then use `qwen_model_provider`. We'll want to do the same here for all models Cerebras supports that match models we already have model profiles for.